api/src/model/request/play_asset_pack_state_request.gd

```diff
@@ -0,0 +1,99 @@
+#	See the License for the specific language governing permissions and
+#	limitations under the License.
+#
+# ##############################################################################
+#
+# Request object that handles asynchronous logic related to 
+# get_asset_pack_state() and provides most updated PlayAssetPackState of a
+# given asset pack.
+# 
+# This object provides relevant getters so that it is possible to retrieve
+# the updated states from this object once the request completes.
+#
+# ##############################################################################
+class_name PlayAssetPackStateRequest
+extends PlayAssetDeliveryRequest
+
+signal request_completed(did_succeed, pack_name, result, exception)
+
+var _pack_name : String
+var _did_succeed : bool
+var _result : PlayAssetPackState
+var _error : PlayAssetPackException
+
+func _init(pack_name):
+	_pack_name = pack_name
+
+# -----------------------------------------------------------------------------
+# Returns the requested asset pack's name
+# -----------------------------------------------------------------------------
+func get_pack_name() -> String:
+	return _pack_name
+
+# -----------------------------------------------------------------------------
+# Returns boolean indicating Request succeeded/failed.
+# -----------------------------------------------------------------------------
+func get_did_succeed() -> bool:
+	return _did_succeed
+
+# -----------------------------------------------------------------------------
+# Returns the result of a succeeded Request, represent by a 
+# CellularDataConfirmationResult enum.
+# -----------------------------------------------------------------------------
+func get_result() -> PlayAssetPackState:
+	return _result
+
+# -----------------------------------------------------------------------------
+# Returns a PlayAssetPackException if Request failed.
+# -----------------------------------------------------------------------------
+func get_error() -> PlayAssetPackException:
+	return _error
+
+# -----------------------------------------------------------------------------
+# Callback functions handling signals emitted from the plugin.
+#
+# Emits request_completed(did_succeed, result, exception) signal upon request 
+# succeeds/fails.
+# 	did_succeed : boolean indicating request succeeded/failed
+# 	pack_name: String, name of the requested asset pack
+# 	result : PlayAssetPackState object if request succeeded, otherwise null
+#	exception: PlayAssetPackException object if request failed, otherwise null
+#
+# Note: when calling get_asset_pack_state() on a non-existent pack_name, 
+# did_succeed will be false and both result and exception will be null.
+# -----------------------------------------------------------------------------
+func _on_get_asset_pack_state_success(result : Dictionary):
+	# getPackStates() in plugin returns a PlayAssetPackStates Dictionary
+	# TODO: look into the exact behavior of get non-existent pack state.
```

Comment:
TODO: look into the exact behavior of get non-existent pack state. 
Currently we are handling as if the plugin won't emit an error, but we will have an empty pack_states Dictionary. Hence if we encounter this situation we would consider it as request failed.